### DISCUSSION TITLE
How to audit sql?

### DISCUSSION THREAD

**Discussion by cutiechi** [apache/kyuubi#2515] [Q&A]
How to send query details to kafka？like https://trino.io/docs/current/develop/event-listener.html

**Comment by wForget** [comment]
There are some events exposed in Kyuubi like: [kyuubi server events](https://github.com/apache/incubator-kyuubi/tree/master/kyuubi-server/src/main/scala/org/apache/kyuubi/events), you can use [ServerJsonLoggingEventHandler](https://github.com/apache/incubator-kyuubi/blob/master/kyuubi-server/src/main/scala/org/apache/kyuubi/events/handler/ServerJsonLoggingEventHandler.scala), or implement a new `EventHandler` to collect these events.

**Reply by cutiechi** [reply]
But I don't know how to configure it, the documentation is too simple

**Reply by wForget** [reply]
The documentation is not yet complete, you can try these two configurations.

Key | Default | Meaning | Type | Since
--- | --- | --- | --- | ---
<code>kyuubi.backend.server.event.json.log.path</code>|<div style='width: 65pt;word-wrap: break-word;white-space: normal'>file:///tmp/kyuubi/events</div>|<div style='width: 170pt;word-wrap: break-word;white-space: normal'>The location of server events go for the builtin JSON logger</div>|<div style='width: 30pt'>string</div>|<div style='width: 20pt'>1.4.0</div>
<code>kyuubi.backend.server.event.loggers</code>|<div style='width: 65pt;word-wrap: break-word;white-space: normal'></div>|<div style='width: 170pt;word-wrap: break-word;white-space: normal'>A comma separated list of server history loggers, where session/operation etc events go.<ul> <li>JSON: the events will be written to the location of kyuubi.backend.server.event.json.log.path</li> <li>JDBC: to be done</li> <li>CUSTOM: to be done.</li></ul></div>|<div style='width: 30pt'>seq</div>|<div style='width: 20pt'>1.4.0</div>
